src/Connection.ts

```diff
@@ -1,1488 +1,1884 @@
+			if (noTimeout !== true) {
+				setTimeout(
+					() =>
+						this.getVersion().then((info) => {
+							const [major, minor, patch] =
+								info.version.split(".");
+							const v =
+								parseInt(major, 10) * 10000 +
+								parseInt(minor, 10) * 100 +
+								parseInt(patch, 10);
+							if (v < 40102) {
+								this._authTimer = null;
+								// possible this is old version of admin
+								this.onPreConnect(false, false);
+							} else {
+								this._socket.emit(
+									"authenticate",
+									(isOk, isSecure) =>
+										this.onPreConnect(isOk, isSecure),
+								);
+							}
+						}),
+					500,
+				);
+			} else {
+				// iobroker websocket waits, till all handlers are installed
+				this._socket.emit("authenticate", (isOk, isSecure) => {
+					this.onPreConnect(isOk, isSecure);
+				});
+			}
+		});
+
+		this._socket.on("reconnect", () => {
+			this.props.onProgress?.(PROGRESS.READY);
+			this.connected = true;
+
+			if (this.waitForRestart) {
+				window.location.reload();
+			} else {
+				this._subscribe(true);
+				this.onConnectionHandlers.forEach((cb) => cb(true));
+			}
+		});
+
+		this._socket.on("disconnect", () => {
+			this.connected = false;
+			this.subscribed = false;
+			this.props.onProgress?.(PROGRESS.CONNECTING);
+			this.onConnectionHandlers.forEach((cb) => cb(false));
+		});
+
+		this._socket.on("reauthenticate", () => this.authenticate());
+
+		this._socket.on("log", (message) => {
+			this.props.onLog?.(message);
+			this.onLogHandlers.forEach((cb) => cb(message));
+		});
+
+		this._socket.on("error", (err: any) => {
+			let _err: string;
+
+			if (err == undefined) {
+				_err = "";
+			} else if (typeof err.toString !== "function") {
+				_err = err.toString();
+			} else {
+				_err = JSON.stringify(err);
+				console.error(`Received strange error: ${_err}`);
+			}
+
+			if (_err.includes("User not authorized")) {
+				this.authenticate();
+			} else {
+				window.alert(`Socket Error: ${err}`);
+			}
+		});
+
+		this._socket.on("connect_error", (err: any) =>
+			console.error(`Connect error: ${err}`),
+		);
+
+		this._socket.on("permissionError", (err) =>
+			this.onError({
+				message: "no permission",
+				operation: err.operation,
+				type: err.type,
+				id: err.id || "",
+			}),
+		);
+
+		this._socket.on("objectChange", (id, obj) => {
+			setTimeout(() => this.objectChange(id, obj), 0);
+		});
+
+		this._socket.on("stateChange", (id, state) => {
+			setTimeout(() => this.stateChange(id, state), 0);
+		});
+
+		this._socket.on("cmdStdout", (id, text) => {
+			this.onCmdStdoutHandler?.(id, text);
+		});
+
+		this._socket.on("cmdStderr", (id, text) => {
+			this.onCmdStderrHandler?.(id, text);
+		});
+
+		this._socket.on("cmdExit", (id, exitCode) => {
+			this.onCmdExitHandler?.(id, exitCode);
+		});
+	}
+
+	/**
+	 * Called internally.
+	 * @param isOk
+	 * @param isSecure
+	 */
+	private onPreConnect(isOk: boolean, isSecure: boolean) {
```

Comment:
We don't care about `isOk`?

Reply:
I don't know - that was already that way in the original code.

Reply:
Ok ;-)